edit.php

```diff
@@ -1,92 +1,50 @@
 <?php
-{
-	mysql_connect('localhost');
-	mysql_select_db('test_db');
+
+	include('header.inc.php');
+
+	if(!isset($_GET['id']) || empty($_GET['id']) || !is_numeric($_GET['id']))
+	{
+		die("Invalid entry ID.");
+	}
+	$id = (int) $_GET['id'];
 
 	if(isset($_POST['update']))
 	{
-		$id = (int)$_GET['id'];
 		$entry = htmlspecialchars(strip_tags($_POST['entry']));
 		$title = htmlspecialchars(strip_tags($_POST['title']));
+		$password = (isset($_POST['password'])) ? 1 : 0;
+		$timestamp = time();
 
-		if(isset($_POST['password']))
-		{
-			$password = 1;
-		}
-		else
-		{
-			$password = 0;
-		}
-
-		$entry = nl2br($entry);
-
-		if(!get_magic_quotes_gpc())
-		{
-			$title = addslashes($title);
-			$entry = addslashes($entry);
-		}
-
-		$timestamp = mktime(date('H'),date('i'),0,date('n'),date('j'),date('Y'));
-		$result = mysql_query("UPDATE php_blog SET timestamp='$timestamp', title='$title', entry='$entry', password ='$password' WHERE id='$id' LIMIT 1") or print("Can't update entry.<br>" . mysql_error());
-		header("Location: view_post.php?id=" . $id);
-
+		$stmt = $database->prepare("UPDATE `php_blog` SET `timestamp`=?, `title`=?, `entry`=?, `password`=? WHERE `id`=? LIMIT 1");
+		$result = $stmt->execute(Array($timestamp, $title, $entry, $password, $id));
+		header("Location: /view_post.php?id=" . $id);
 	}
 
 	if(isset($_POST['delete']))
 	{
-		$id = (int)$_GET['id'];
-		$result = mysql_query("DELETE FROM php_blog where id = '$id'") or print("Can't delete entry. <br>" . mysql_error());
-
-		if($result != false)
-		{
-			header("Location: index.php");
-		}
+		$stmt = $database->prepare("DELETE FROM `php_blog` where `id`=?");
+		$result = $stmt->execute(Array($id));
+		header( "Location: /index.php" );
 	}
 
-	if(!isset($_GET['id']) || empty($_GET['id']) || !is_numeric($_GET['id']))
-	{
-		die("Invalid entry ID.");
-	}
-	else
-	{
-		$id = (int)$_GET['id'];
-	}
```

Comment:
You do this in both of the ifs above, so you might as well just do it once up the top of the file.